src/FormattingSettingsService.ts

```diff
@@ -50,74 +53,103 @@ export class FormattingSettingsService implements IFormattingSettingsService {
+                            formattingGroup = (<Group>cardGroupInstance).getFormattingGroup(this.localizationManager);
+                            formattingCard.groups.push(formattingGroup);
+                        }
+                        
+                        // In case formatting model adds data points or top categories (Like when you modify specific visual category color).
+                        // these categories use same object name and property name from capabilities and the generated uid will be the same for these formatting categories properties
+                        // Solution => Save slice names to modify each slice uid to be unique by adding counter value to the new slice uid
+                        const sliceNames: { [name: string]: number } = {};
+
+                        // Build formatting container slice for each property
+                        if (cardGroupInstance.container) {
+                            const container = cardGroupInstance.container;
+                            const containerUid = groupUid + "-container";
+                            const formattingContainer: visuals.FormattingContainer = {
+                                displayName: (this.localizationManager && container.displayNameKey)
+                                    ? this.localizationManager.getDisplayName(container.displayNameKey) : container.displayName,
+                                description: (this.localizationManager && container.descriptionKey)
+                                    ? this.localizationManager.getDisplayName(container.descriptionKey) : container.description,
+                                containerItems: [],
+                                uid: containerUid
+                            }
+
+                            container.containerItems.forEach((containerItem: formattingSettings.ContainerItem) => {
+                                // Build formatting container item object
+                                const containerIemName = containerItem.displayNameKey ? containerItem.displayNameKey : containerItem.displayName;
+                                const containerItemUid: string = containerUid + containerIemName;
+                                let formattingContainerItem: visuals.FormattingContainerItem = {
+                                    displayName: (this.localizationManager && containerItem.displayNameKey)
+                                        ? this.localizationManager.getDisplayName(containerItem.displayNameKey) : containerItem.displayName,
+                                    slices: [],
+                                    uid: containerItemUid
+                                }
+
+                                // Build formatting slices and add them to current formatting container item
+                                this.buildFormattingSlices(containerItem.slices, objectName, sliceNames, index === 0 ? formattingCard : formattingGroup, formattingContainerItem.slices);
```

Comment:
The use here of "index === 0" and in line 136, will be wrong and cause a bug in case a card have one/multiple groups
and the first group has topLevelToggle, then the topLevelToggle will be added to the card instead of the group

Please replace of index==0 by 
formattingGroup.displayName==undefined ? formattingCard : formattingGroup

Reply:
See https://github.com/microsoft/powerbi-visuals-utils-formattingmodel/pull/7#discussion_r1147639170 for the current implementation of the `topLevelToggle` and why `index === 0 ? formattingCard : formattingGroup` selects the correct parent for the `topLevelToggle`

Reply:
please see my comment in that conversation again